src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,191 @@
+
+#### Scenario 1: `amplify push` fails after manually updating DynamoDB tables and GSIs.
+Assume a situation where you have renamed GraphQL [@model](https://docs.amplify.aws/cli/graphql/data-modeling/)s in your Amplify CLI project and updated their [@index](https://docs.amplify.aws/cli/graphql/data-modeling/) parameters. 
+However, at this point you use the DynamoDB console in your account to delete multiple [GSIs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html) belonging to the Amplify application which belonged to the older model names. 
+This action will introduce drift in the CloudFormation stack deployed by Amplify. Since the GSIs corresponding to the older DynamoDB tables are deleted, any change to index parameters will fail during deployment. 
+
+##### Suggested Resolution:
+To fix deployment failures due to drift, manually rollback the state of the drifted resource to match its state in CloudFormation. Then retry `amplify push`. 
+1. You can view the drift in the DynamoDB CloudFormation Stack per instructions in the [AWS Drift Detection document](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/detect-drift-stack.html)
+1. Revert every drift manually through the AWS console. In this particular scenario, re-create the GSIs and manually undo any renaming changes.
+1. Perform an `amplify push`.
+
+#### Scenario 2: `amplify push` fails after manually updating (non database) resources in the cloud.
+
+* You have manually updated your GraphQL schema using the AppSync console. Or
+* You have manually deleted user pools from the Cognito console. Or
+* You have updated the code in Lambda functions which were originally deployed through Amplify.
+
+##### Suggested Resolution: 
+These types of drift must be addressed on a case-by-case basis. In some cases, such as a Lambda code change, forcing a redeployment of the function by making a minor change to the code may work. In other cases, like deleting user pools or updates to GraphQL schema from the AppSync console, try the following steps:
+1. Forcible deployment of CloudFormation stack from the CLI - Run `amplify push --force` to force push the application’s resource stack to be deployed.
+1. Forcible deployment of CloudFormation stack from the Console - Fetch the CloudFormation stack from the application’s deployment bucket and selectively run the resource specific part of the stack through the CloudFormation console.
+1. Apply drift changes into code: If both the above methods to deploy CloudFormation have failed for you, the other option is to inspect the manual changes and to update them in the CLI input files. For example, you could update the GraphQL schema to reflect the same changes as performed manually in the cloud.
+
+### Push failures in a multi-account multi-environment project (CI/CD errors)
```

Comment:
Do we have the error codes that are thrown here? I like how it was incorporated in other sections above.